modules/module.c

```diff
@@ -112,29 +116,25 @@ static int import(size_t argc, struct cb_value *argv, struct cb_value *result)
 		}
 	} else {
 		/* FIXME: Keep track of current file directory so pwd can be set */
-		f = cb_agent_resolve_import(import_name, NULL, &path);
+		f = cb_agent_resolve_import(import_name_str, NULL, &path);
 		if (!f)
 			goto err;
 	}
 
-	pc = cb_bytecode_len(cb_vm_state.bytecode);
-	if (cb_compile_module(cb_vm_state.bytecode, import_name, f, path))
+	import_name = cb_agent_intern_string(cb_strptr(&import_name_str),
+			cb_strlen(import_name_str));
+	modspec = cb_agent_get_modspec_by_name(import_name);
+	if (!modspec) {
+		modspec = cb_modspec_new(import_name);
+		cb_agent_add_modspec(modspec);
+	}
+
+	code = cb_compile_file(modspec, f);
```

Comment:
would be nice to avoid compiling and evaluating again if the module has already been imported